kube_controller_manager/tests/test_e2e.py

```diff
@@ -0,0 +1,15 @@
+import pytest
+
+from datadog_checks.base import AgentCheck
+
+
+@pytest.mark.e2e
+def test_e2e(dd_agent_check, aggregator, instance):
+
+    with pytest.raises(Exception):
+        dd_agent_check(instance, rate=True)
+    endpoint_tag = "endpoint:" + instance.get('prometheus_url')
+    tags = instance.get('tags').append(endpoint_tag)
```

Comment:
Note: be careful because you are modifying the instance after it's used by the check. It's not a problem here since you do not reuse the instance but it could lead to issues if you use the instance again. Feel free to keep it as it is 🙂

Reply:
tbh I'd vote just as a matter of habit not to modify the instance. no knowing when this may bite us in the back.

Especially since a) you have to modify that line anyway (`dict.get` will return `None` for missing keys which will crash when you call `append` on it) and b) the alternative is very simple too:
```
tags = instance.get('tags', []) + [endpoint_tag]
```